travis/pre_commit.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+echo ""
+echo "===================================================="
+echo "Running this library's configured pre-commit checks."
+echo "See https://pre-commit.com/ for more information."
+echo "===================================================="
+echo ""
+
+pip install pre-commit
+pre-commit install
+
+pushd "${TRAVIS_BUILD_DIR}"
+pre-commit run --all-files
+PRE_COMMIT_EXIT_CODE=$?
+
+if [ $PRE_COMMIT_EXIT_CODE -ne 0 ]; then
+    echo ""
+    echo "====================================================================="
+    echo "WARNING: One or more pre-commit checks have failed!"
+    echo "This means you do not have pre-commit set up in your local checkout!"
```

Comment:
Seems like it's "maybe not set up or tests failed"?

Reply:
If pre-commit was set up correctly locally and you still managed to push a bad change, then something is very wrong. Maybe this should be reworded to reflect this.

Reply:
Oh, wait, I think I misunderstood the intent of this message. Feel free to disregard.